### DISCUSSION TITLE
Maintenance announcements

### DISCUSSION THREAD

**Discussion by qgabrel** [netbox-community/netbox#9691] [Ideas]
Hello,

I am looking for a way to send emails to my customers for maintenance announcements.

Example: I need to upgrade a PE, I select this device, and Netbox knows all the customer services related to it. As Netbox knows the services, it also knows the customer. All that remains is for the software to send an email to the customer contacts.

We can imagine some communication templates to avoid to change the text every time.

What do you think about it ?

**Comment by DanSheps** [comment]
You would want to write a plugin or script to handle this.

**Comment by qgabrel** [comment]
Thanks for feedback. 
I'm pretty sure it should help some people. Any idea to do this ?

**Reply by jeremystretch** [reply]
Our [plugin tutorial](https://github.com/netbox-community/netbox-plugin-tutorial) is probably the best way to get started. It covers all the major aspects of building a plugin.
